app/controllers/authentications_controller.rb

```diff
@@ -33,7 +33,7 @@ def destroy
     if @authentication
       if current_user.authentications.count == 1 and current_user.encrypted_password.blank?
         # Bryan: TESTED
-        flash[:alert] = 'Bad idea!'
+        flash[:alert] = 'Failed to unlink GitHub. Please use another provider for login or reset password.'
```

Comment:
The detailed message did came in my mind, however i wanted to keep message size as minimum as possible with still delivering the intended message. Keyword `provider` was used to keep it generic, just in case if we decide to add one more provider.